lib/kubernetes-deploy/task_config_validator.rb

```diff
@@ -11,12 +11,12 @@ class TaskConfigValidator
 
     delegate :context, :namespace, :logger, to: :@task_config
 
-    def initialize(task_config, kubectl, kubeclient_builder, only: nil)
+    def initialize(task_config, kubectl, kubeclient_builder, only: nil, skip: [])
       @task_config = task_config
       @kubectl = kubectl
       @kubeclient_builder = kubeclient_builder
       @errors = nil
-      @validations = only || DEFAULT_VALIDATIONS
+      @validations = (only || DEFAULT_VALIDATIONS) - skip
```

Comment:
I'm kinda surprised to see that in addition to subclassing, we also need the ability to modify the defaults with both `only` and `skip` options. Am I reading correctly that the `only` option is just used by tests? Could this work:

```ruby
class TaskConfigValidator
  def validations
    DEFAULT_VALIDATIONS
  end
end

class SomeValidator < TaskConfigValidator
  def validations
    DEFAULT_VALIDATIONS - [:validation_to_skip] # or + something, or just a specific list
  end
end
```

Then the superclass isn't messing around with constructing the list, when it should in practice be a static property of each subclass.